tests/unit/mathematics/SGLinalg_unittest.cc

```diff
@@ -60,6 +60,19 @@ TEST(SGLinalg, CPUBackend_dot)
 	EXPECT_NEAR(result, 20.0, 1E-15);
 }
 
+TEST(SGLinalg, CPUBackend_sum)
+{
+	const index_t size=10;
```

Comment:
This is not an optimal unit test.
Why don't you use range_fill, then at least we guard against a few more cases